### PR TITLE
Laboratory work #1, Darya Balba - 21FPL2

### PR DESCRIPTION
**Pull request by daryabalba**

### CODE REVIEW

**Review by artyomtugaryov — COMMENTED** [review]
Very good! Please, call the implemented functions in start.py

**Review by artyomtugaryov — APPROVED** [review]

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -66,7 +98,21 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not frequencies or not top:
+        return None
+    if not type(top).__name__ == 'int' or not isinstance(frequencies, dict):
```
[the remark this answers is not in the document]
Comment (daryabalba) [reply lab_1_keywords_tfidf/main.py:103]: Done

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -99,7 +159,25 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
+            if not isinstance(key, str) or not isinstance(value, float):
+                return None
+    for key in idf.keys():
+        for value in idf.values():
+            if not isinstance(key, str) or not isinstance(value, float):
+                return None
+    new = {}
+    for term_key in term_freq.keys():
+        if term_key not in idf:
+            new[term_key] = term_freq[term_key] * math.log(47/1)
```
[the remark this answers is not in the document]
Comment (daryabalba) [reply lab_1_keywords_tfidf/main.py:177]: Done

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -66,7 +98,21 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
...
+        return None
+    for i in frequencies.keys():
+        if not isinstance(i, str):
+            return None
+    for j in frequencies.values():
+        if not isinstance(j, int) and not isinstance(j, float):
+            return None
+    sorted_dict = dict(sorted(frequencies.items(), key=lambda x: x[1], reverse=True))
+    sorted_top = []
+    for i in sorted_dict.keys():
```
[the remark this answers is not in the document]
Comment (daryabalba) [reply lab_1_keywords_tfidf/main.py:113]: Done

**Pull request closed without merging**